examples/asr/transcribe_speech.py

```diff
@@ -116,6 +116,9 @@ class TranscriptionConfig:
     # Decoding strategy for RNNT models
     rnnt_decoding: RNNTDecodingConfig = RNNTDecodingConfig(fused_batch_size=-1)
 
+    # This will be merged with the model config, overriding existing parameters.
+    merge_into_model_config: Optional[Any] = None
```

Comment:
I do not think we should add such an argument. Merging of configs is a very flaky feature, if you try to add keys that didn't exist in the main config it will fail, it requires subconfig knowledge. 

In the name of making it simpler we are adding loads of complexity + chance of failure. 

Users can and see recommended to use the restore_config=True, update config, then override_cfg workflow. 

Please revert the changes wrt partial merge.

Reply:
Reverted, added instead method for changing model in utils